pkg/products/marin3r/reconciler.go

```diff
@@ -220,6 +295,10 @@ func (r *Reconciler) preUpgradeBackupExecutor() backup.BackupExecutor {
 	if r.installation.Spec.UseClusterStorage != "false" {
 		return backup.NewNoopBackupExecutor()
 	}
-	//todo add backup for redis once it's added to the reconciler
-	return backup.NewNoopBackupExecutor()
+
+	return backup.NewAWSBackupExecutor(
+		r.installation.Namespace,
+		externalRedisSecretName,
```

Comment:
```suggestion
		fmt.Sprintf("%s%s", constants.RateLimitRedisPrefix, r.installation.Name),
```

This parameter is the name of the Redis CR, which is referenced by the RedisSnapshot created when running the backups

Reply:
Good catch, thanks